src/main/java/journeymap/client/data/WorldData.java

```diff
@@ -285,10 +287,25 @@ public static String getSafeDimensionName(WorldProvider worldProvider)
 
         try
         {
-            return worldProvider.getDimensionName();
+            if (dimNames == null) {
+                dimNames = new TIntObjectHashMap<String>();
```

Comment:
let's create this in a static block below the constructor, this way it is created as soon as the class is loaded, no need for the null check. 
   
```
    /**
     * Constructor.
     */
    public WorldData()
    {
    }
    
    static 
    {
        dimNames = new TIntObjectHashMap<>();
    }

```